_posts/2018-08-20-grpc_on_http2.md

```diff
@@ -0,0 +1,76 @@
+---
+layout: post
+title: gRPC on HTTP/2: Engineering a Robust, High Performance Protocol
+published: true
+permalink: blog/grpc_on_http2
+author: Jean de Klerk
+author-link: https://github.com/jadekler
+company: Google
+company-link: https://www.google.com
+---
+In a [previous article](https://grpc.io/blog/http2_smarter_at_scale), we explored how HTTP/2 dramatically increases network efficiency and enables real-time communication by providing a framework for long-lived connections. In this article, we’ll look at how gRPC builds on HTTP/2’s long-lived connections to create a performant, robust platform for inter-service communication. [gRPC](http://grpc.io/) is a high-performance, open-source universal RPC framework. We will explore the relationship between gRPC and HTTP/2, how gRPC manages HTTP/2 connections, and how gRPC uses HTTP/2 to keep connections alive, healthy, and utilized.
+<!--more-->
+
+## gRPC Semantics
+
+To begin, let’s dive into how gRPC concepts relate to HTTP/2 concepts. gRPC introduces three new concepts: *channels* [1], *remote procedure calls* (RPCs), and *messages*. The relationship between the three is simple: each channel may have many RPCs while each RPC may have many messages.
+
+<img src="https://grpc.io/img/channels_mapping_2.png" alt="Channel Mapping" style="max-width: 800px">
+
+Let’s take a look at how gRPC semantics relate to HTTP/2:
+
+<img src="https://grpc.io/img/grpc_on_http2_mapping_2.png" alt="gRPC on HTTP/2" style="max-width: 800px">
+
+Channels are a key concept in gRPC. Streams in HTTP/2 enable multiple concurrent conversations on a single connection; channels extend this concept by enabling multiple streams over multiple concurrent connections. On the surface, channels provide an easy interface for users to send messages into; underneath the hood, though, an incredible amount of engineering goes into keeping these connections alive, healthy, and utilized.
```

Comment:
nit: "interface for users to send messages into"

This is like ending a sentence with a preposition, it's a jarring end to the statement.

Reply:
ack